### PR TITLE
Put uchiwa behind Apache

### PR DESCRIPTION
**Pull request by larsks**
I am proposing this as alternative to #62.

This change configures both uchiwa and kibana to run behind an Apache frontend, and enforces SSL access to the web UI.  Both applications can be run on the same host (or on different hosts).  In addition to changes to the uchiwa, kibana, and httpd roles, this introduces a new "opstoolsvhost" role that is responsible for a top-level Apache configuration file.

This does not use the generalized proxy support proposed in #62 because the applications each required unique proxy configurations.

### CODE REVIEW

**Comment by larsks** [comment]
Incidentally, I've tested this with all services on a single host:

```
[logging_hosts]
server0

[am_hosts]
server0
```

And with logging and monitoring services separate:

```
[logging_hosts]
server0

[am_hosts]
server1
```

Note that there are still some outstanding issues with the default ip addresses we use for binding and connecting, but I would like to make that part of a separate PR.

File: roles/opstoolsvhost/templates/opstools.conf.j2 [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+<VirtualHost *:{{ opstools_apache_http_port }}>
+  RewriteEngine On
+  RewriteCond %{HTTPS} off
+  RewriteRule (.*) https://%{HTTP_HOST}$1 [R,L]
+</VirtualHost>
+
+<VirtualHost *:{{ opstools_apache_https_port }}>
+  RewriteEngine On
+  ProxyRequests Off
+  SSLEngine On
```
Comment (mrunge) [inline roles/opstoolsvhost/templates/opstools.conf.j2:17]: if mod_ssl is not loaded, this will fail here.

**Review by mrunge — CHANGES_REQUESTED** [review]
nearly there, make sure mod_ssl is loaded, when configuring ssl

File: roles/opstoolsvhost/templates/opstools.conf.j2 (hunk 1 above)
Comment (mrunge) [reply roles/opstoolsvhost/templates/opstools.conf.j2:17]: rewrite engine needs to be loaded too

Comment (mrunge) [reply roles/opstoolsvhost/templates/opstools.conf.j2:17]: one more: package mod_ssl needs to be installed. I tested with a fresh and minimal CentOS 7 image

Comment (jbadiapa) [reply roles/opstoolsvhost/templates/opstools.conf.j2:17]: I also had to seboolean httpd_can_network_connect on

**Comment by jbadiapa** [comment]
The role/httpd/proxy is no longer used , I think it should be deleted.

File: roles/opstoolsvhost/templates/opstools.conf.j2 (hunk 1 above)
Comment (larsks) [reply roles/opstoolsvhost/templates/opstools.conf.j2:17]: I'll update these as soon as I get my kids to school.

**Review by mrunge — APPROVED** [review]
thanks, looks better now. I could verify it works. Let's fix the connection issues with elastic in a different patch.

**Comment by jbadiapa** [comment]
+1, It works for me

**Review by paramite — APPROVED** [review]

**Pull request merged**